### PR TITLE
Use kernel messages provider from utils

### PR DESCRIPTION
**Pull request by d3dave**

### CODE REVIEW

File: gprofiler/kernel_messages.py [hunk 1]
```diff
@@ -2,41 +2,32 @@
...
-
-class EmptyProvider(Provider):
-    def __init__(self):
-        print(
-            "This kernel does not support the new /dev/kmsg interface for reading messages,"
-            " or you lack the permissions for it."
-        )
+def get_kernel_messages_provider():
+    if get_kernel_release() < (3, 5):
+        print("This kernel does not support the new /dev/kmsg interface for reading messages.")
```
Comment (Jongy) [inline gprofiler/kernel_messages.py:20]: Btw, why prints and not logger?

File: gprofiler/kernel_messages.py [hunk 2]
```diff
@@ -2,41 +2,32 @@
...
-if kernel_release >= (3, 5):
-    from gprofiler.devkmsg import DevKmsgProvider
-
-    DefaultMessagesProvider = DevKmsgProvider
-else:
-    DefaultMessagesProvider = EmptyProvider
+    try:
+        return GProfilerKernelMessagesProvider()
+    except Exception:
+        print("Failed to start kernel messages listener. Do you have permission to read /dev/kmsg ?")
```
Comment (Jongy) [inline gprofiler/kernel_messages.py:28]: Again why prints? Best just with logger I suppose

**Comment by Jongy** [comment]
Btw, it's easier if you push to the public repo, as it makes it easier for others to check out the code locally (now I have to add a new remote / use github's git commands to check our the code)

File: gprofiler/kernel_messages.py [hunk 3]
```diff
@@ -2,41 +2,32 @@
...
-        print(
-            "This kernel does not support the new /dev/kmsg interface for reading messages,"
-            " or you lack the permissions for it."
-        )
+def get_kernel_messages_provider():
+    if get_kernel_release() < (3, 5):
+        print("This kernel does not support the new /dev/kmsg interface for reading messages.")
         print("Profilee error monitoring not available.")
         print()
         logger.warning("Profilee error monitoring not available.")
```
Comment (Jongy) [inline gprofiler/kernel_messages.py:23]: Add `return EmptyKernelMessagesProvider()` here? So it doesn't fall into the `try`

**Review by Jongy — CHANGES_REQUESTED** [review]
A few small comments. I will next look at the code in granulate-utils

File: gprofiler/kernel_messages.py [hunk 4]
```diff
@@ -2,41 +2,32 @@
...
 
-
-class EmptyProvider(Provider):
-    def __init__(self):
-        print(
-            "This kernel does not support the new /dev/kmsg interface for reading messages,"
-            " or you lack the permissions for it."
-        )
+def get_kernel_messages_provider():
+    if get_kernel_release() < (3, 5):
```
Comment (Jongy) [inline gprofiler/kernel_messages.py:19]: Shouldn't this logic move as well? Since it's going to be needed by whomever is going to use this API.

Comment (Jongy) [reply gprofiler/kernel_messages.py:19]: Oh, I see that you have in https://github.com/Granulate/granulate-utils/pull/11. It is still cloned however.. Just for the logging?

Comment (d3dave) [reply gprofiler/kernel_messages.py:19]: yes

Comment (Jongy) [reply gprofiler/kernel_messages.py]: Why removed tho? I think the "No kernel messages monitoring" message is nice.

Comment (d3dave) [reply gprofiler/kernel_messages.py]: readded

**Comment by Jongy** [comment]
Tested again?

**Comment by d3dave** [comment]
Tested:

```
[15:14:50] Profiling process 14961 with async-profiler
[15:14:53] Async-profiled Java process exited with error code (pid=14961, exit_code=0x9)
[15:14:53] Profiled Java process OOM (oom={"timestamp": null, "message": null, "pid": 14961, "comm": "java", "total_vm": 12480106496, "anon_rss": 6934462464, "file_rss": 1511424, "shmem_rss": 0})
[15:14:53] Java profiling has been disabled, will avoid profiling any new java process
```

**Comment by Jongy** [comment]
> [15:14:53] Profiled Java process OOM (oom={"timestamp": null, "message": null, "pid": 14961, "comm": "java", "total_vm": 12480106496, "anon_rss": 6934462464, "file_rss": 1511424, "shmem_rss": 0})

What was the full message line that yielded `null` message?

**Comment by Jongy** [comment]
Merge the utils PR https://github.com/Granulate/granulate-utils/pull/11 then we can advance with this

**Comment by d3dave** [comment]
> Merge the utils PR [Granulate/granulate-utils#11](https://github.com/Granulate/granulate-utils/pull/11) then we can advance with this

Merged.

> > [15:14:53] Profiled Java process OOM (oom={"timestamp": null, "message": null, "pid": 14961, "comm": "java", "total_vm": 12480106496, "anon_rss": 6934462464, "file_rss": 1511424, "shmem_rss": 0})
> 
> What was the full message line that yielded `null` message?

```
3,63387,559650997372,-;Killed process 14961 (java) total-vm:12187604kB, anon-rss:6771936kB, file-rss:1476kB, shmem-rss:0kB
```

**Comment by Jongy** [comment]
Hmm. I suppose `oom_kill_process` got called with an empty message.

**Comment by Jongy** [comment]
Upgrade submodule revision to the merged one?

**Comment by d3dave** [comment]
> Upgrade submodule revision to the merged one?

Done

**Review by Jongy — APPROVED** [review]

**Pull request merged**